lucene/core/src/java/org/apache/lucene/search/AbstractKnnVectorQuery.java

```diff
@@ -156,6 +196,44 @@ private TopDocs getLeafResults(
     }
   }
 
+  private DocIdSetIterator executeSeedQuery(LeafReaderContext ctx, Weight seedWeight)
+      throws IOException {
+    if (seedWeight != null) {
+      // Execute the seed query
+      TopScoreDocCollector seedCollector =
+          new TopScoreDocCollectorManager(k, Integer.MAX_VALUE).newCollector();
```

Comment:
Wondering if we could use a lower `totalHitsThreshold` here since `seedCollector.topDocs().totalHits` seems to be unused?
```suggestion
          new TopScoreDocCollectorManager(k, k).newCollector();
```